src/react-apps/applications/runtime/src/components/base/ButtonComponent.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import FormDataActions from './../../../src/features/form/data/actions/index';
+import { IAltinnWindow } from './../../types';
+export interface IButtonProps {
+  id: string;
+  text: string;
+  disabled: boolean;
+  handleDataChange: (value: any) => void;
+  unsavedChanges: boolean;
+  formDataCount: number;
+  validationResults: any;
+}
+
+export interface IButtonState { }
+
+export class ButtonComponent extends React.Component<IButtonProps, IButtonState> {
+  public renderSubmitButton = () => {
+    const disabled = (this.props.formDataCount > 0 &&
+      (this.props.validationResults !== null && Object.keys(this.props.validationResults).length !== 0))
+      || this.props.unsavedChanges || this.props.formDataCount === 0;
+    return (
+      <button
+        type='submit'
+        className={disabled ? 'a-btn a-btn-success disabled' : 'a-btn a-btn-success'}
+        onClick={this.submitForm}
+        disabled={disabled}
+        id={this.props.id}
+      >
+        {this.props.text}
+      </button>
+    );
+  }
+
+  public renderSaveButton = () => {
+    const disabled = !this.props.unsavedChanges;
+    return (
+      <button
+        type='submit'
+        className={disabled ?
+          'a-btn a-btn-success disabled' : 'a-btn a-btn-success'}
+        onClick={this.saveFormData}
+        disabled={disabled}
+        id='saveBtn'
+      >
+        Lagre
```

Comment:
Hardcoded?

Reply:
Keeping this, as we will remove the save part of the button once we have implemented automatic saving of form.